easybuild/easyblocks/generic/cmakemake.py

```diff
@@ -104,8 +133,27 @@ def configure_step(self, srcdir=None, builddir=None):
 
         options = ['-DCMAKE_INSTALL_PREFIX=%s' % self.installdir]
 
-        if self.cfg['build_type'] is not None:
-            options.append("-DCMAKE_BUILD_TYPE=%s" % self.cfg['build_type'])
+        if '-DCMAKE_BUILD_TYPE=' in self.cfg['configopts']:
+            if self.cfg['build_type'] is not None:
+                self.log.warning('CMAKE_BUILD_TYPE is set in configopts. Ignoring build_type')
+        else:
+            options.append('-DCMAKE_BUILD_TYPE=%s' % self.build_type)
+
+        # Add -fPIC flag if necessary
+        if self.toolchain.options['pic']:
+            options.append('-DCMAKE_POSITION_INDEPENDENT_CODE=ON')
+
+        # Set flag for shared libs if requested
+        # Not adding one allows the project to choose a default
+        build_shared_libs = self.cfg['build_shared_libs']
+        if build_shared_libs is not None:
+            # Contrary to other options build_shared_libs takes precedence over configopts.
+            # This is to allow self.lib_ext to be determined correctly.
+            # Notify users about this potentially unexpected behavior,
+            # they can set `build_shared_libs = None` (default) in the EC to avoid this.
```

Comment:
The recommendation in the comment should be to remove `-DBUILD_SHARED_LIBS=` from `configopts` in the easyconfig file, and use `build_shared_libs = True` or `build_shared_libs = False` instead?

Reply:
Updated in original commit: https://github.com/easybuilders/easybuild-easyblocks/pull/1929/commits/8650cc569abe0d66ffcf1d6e6db83f75f57a532f